Objects/listobject.c

```diff
@@ -2649,6 +2675,15 @@ list___init___impl(PyListObject *self, PyObject *iterable)
         (void)_list_clear(self);
     }
     if (iterable != NULL) {
+        if(_PyObject_HasLen(iterable)){
+            Py_ssize_t iter_len = PyObject_Length(iterable);
+            if (iter_len == -1) {
+                PyErr_Clear();
```

Comment:
I'm not comfortable with silencing all exceptions. It can be a KeyboardInterrupt for example.